metrics/storage/storage.go

```diff
@@ -32,6 +32,20 @@ var (
 	limiter = rate.NewLimiter(50, 50)
 )
 
+type Limiter interface {
```

Comment:
What's this needed for? You could just pass around a maybe-nil `*rate.Limiter` ?

Reply:
It defines a more constrained interface that can do nothing but `Wait()`.

Reply:
Probably comment:
// More simple subset of the rate.Limiter capabilities